tensorflow/lite/java/src/main/java/org/tensorflow/lite/Tensor.java

```diff
@@ -313,7 +313,7 @@ private void throwIfTypeIsIncompatible(Object o) {
     DataType oType = dataTypeOf(o);
 
     // INT8 and UINT8 have the same string name, "byte"
-    if (oType.toStringName() == dtype.toStringName()) {
+    if (oType.toStringName().equals(dtype.toStringName())) {
```

Comment:
Thanks! And to avoid  unnecessary comparisons, I would move this check underneath the oType != dtype conditional body.

Reply:
Oops, done.